prqlc/prqlc/Cargo.toml

```diff
@@ -62,6 +63,7 @@ strum_macros = "0.26.2"
 anyhow = {version = "1.0.82", features = ["backtrace"], optional = true}
 clap = {version = "4.4.18", features = ["derive", "env", "wrap_help"], optional = true}
 clap_complete_command = {version = "0.5.1", optional = true}
+clap-verbosity-flag = { version = "2.2.0", optional = true }
```

Comment:
The addition of this seems to bring the MSRV to 1.74.
My question is that changing the MSRV of prqlc here will even affect its functionality as a built-in that does not require a
cli feature.
Why are cli features included in default? I think it would be better to exclude them from default.

(As a side note, I think the MSRV could be 1.74 for now. I just don't think it should be pulled to other parts of the cli feature about the future.)

Reply:
@m-span if we downgrade to 2.1.2 then we get the 1.70.0 MSRV; let's do that I think. Then we'll upgrade when MSRVs match in the near future.

Reply:
> IIUC the R ecosystem & debian were the main impediments to increasing the MSRV; is that still the case?

Over the past month, rustc on Debian has migrated rapidly from 1.70 to 1.74, and Ubuntu is currently at 1.75, so I consider it quite unlikely that setting 1.74 or 1.75 will cause problems.

![image](https://github.com/PRQL/prql/assets/50911393/a8c6973a-a78f-4e3c-829a-5ae588d216f5)

The problem with CRAN is that it continues to use Fedora 36 (time stopped at rustc 1.69), which reached EOL a year ago, so it is not worth considering.
They do not have a roadmap for how long they will continue to use Fedora 36 and cannot predict the future.

Since they are volunteers and do not seem to have the resources to update Fedora under the current system, I suspect that this situation may continue until the infrastructure is completely rebuilt on an IaC basis.

Either way I will either patch to lower the MSRV (which may involve manual modification of the lock file and is a tremendous pain) or simply give up submitting to CRAN.

Reply:
Ah OK, that's a shame re CRAN, hope they figure out a way to go forwards.

Given they're at 1.69, I guess staying on 1.70 doesn't help much? And we can bump to 1.74? I think it's quite easy for us to stay at 1.70 (in this case a tiny downgrade), but sounds like that's not actually that helpful for CRAN?

Reply:
Thanks, I think the process of lowering from 1.70 to 1.69 is much easier than the process of lowering from 1.74 to 1.69.
As far as I am aware there is no new syntax introduced in 1.70 in prqlc at the moment, just some dependencies.
If the checks in 1.70 were to be stopped, it would include many more dependencies, which would make it more difficult to identify the problem.

Reply:
OK, let's leave it at 1.70 for the moment then.

Let's do the downgrade and merge this PR!

Reply:
Apparently we aren't even on MSRV 1.70 right now:

```
~/projects/prql git:[main]
cargo msrv --linear
Fetching index
Determining the Minimum Supported Rust Version (MSRV) for toolchain aarch64-apple-darwin
Using check command cargo check
Check for toolchain '1.78.0-aarch64-apple-darwin' succeeded
Check for toolchain '1.77.2-aarch64-apple-darwin' succeeded
Check for toolchain '1.76.0-aarch64-apple-darwin' succeeded
Check for toolchain '1.75.0-aarch64-apple-darwin' succeeded
Check for toolchain '1.74.1-aarch64-apple-darwin' succeeded

Check for toolchain '1.73.0-aarch64-apple-darwin' failed with:
┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ error: package `pulldown-cmark v0.10.3` cannot be built because it requires rustc 1.74 or newer, while the currently active rustc version is 1.73.0                           │
│ Either upgrade to rustc 1.74 or newer, or use                                                                                                                                 │
│ cargo update -p pulldown-cmark@0.10.3 --precise ver                                                                                                                           │
│ where `ver` is the latest version of `pulldown-cmark` supporting rustc 1.73.0                                                                                                 │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
   Finished The MSRV is: 1.74.1   ██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ 00:00:03
   ```

Reply:
but yes, I did downgrade that package to `2.1.2`

Reply:
> Apparently we aren't even on MSRV 1.70 right now:

Please check CI, only prqlc is confirming MSRV.
https://github.com/PRQL/prql/blob/045c03dfc4f28a0e484e50f0ad6ba0d8e7c09689/.github/workflows/tests.yaml#L495-L498

> but yes, I did downgrade that package to `2.1.2`

The lock file does not seem updated.